spec/lib/annotate/annotate_models_spec.rb

```diff
@@ -954,283 +954,429 @@ def mock_column(name, type, options = {})
+                #  notes  :text(55)         not null
+                #
+              EOS
+            end
 
-      it 'should get schema info as RDoc' do
-        klass = mock_class(:users,
-                           :id,
-                           [
-                             mock_column(:id, :integer, comment: 'ID'),
-                             mock_column(:name, :string, limit: 50, comment: 'Name')
-                           ])
-        expect(AnnotateModels.get_schema_info(klass, AnnotateModels::PREFIX, format_rdoc: true, with_comment: true)).to eql(<<-EOS.strip_heredoc)
-        # #{AnnotateModels::PREFIX}
-        #
-        # Table name: users
-        #
-        # *id(ID)*::     <tt>integer, not null, primary key</tt>
-        # *name(Name)*:: <tt>string(50), not null</tt>
-        #--
-        # #{AnnotateModels::END_MARK}
-        #++
-        EOS
+            it 'works with option "hide_limit_column_types"' do
+              is_expected.to eq expected_result
+            end
+          end
+
+          context 'when "hide_limit_column_types" is "integer,boolean,string,text"' do
+            let :options do
+              { hide_limit_column_types: 'integer,boolean,string,text' }
+            end
+
+            let :expected_result do
+              <<~EOS
+                # Schema Info
+                #
+                # Table name: users
+                #
+                #  id     :integer          not null, primary key
+                #  active :boolean          not null
+                #  name   :string           not null
+                #  notes  :text             not null
+                #
+              EOS
+            end
+
+            it 'works with option "hide_limit_column_types"' do
+              is_expected.to eq expected_result
+            end
+          end
+        end
+
+        context 'when "hide_default_column_types" is specified in options' do
+          let :columns do
+            [
+              mock_column(:profile, :json, default: {}),
+              mock_column(:settings, :jsonb, default: {}),
+              mock_column(:parameters, :hstore, default: {})
+            ]
+          end
+
+          context 'when "hide_default_column_types" is blank string' do
+            let :options do
+              { hide_default_column_types: '' }
+            end
+
+            let :expected_result do
+              <<~EOS
+                # Schema Info
+                #
+                # Table name: users
+                #
+                #  profile    :json             not null
+                #  settings   :jsonb            not null
+                #  parameters :hstore           not null
+                #
+              EOS
+            end
+
+            it 'works with option "hide_default_column_types"' do
+              is_expected.to eq expected_result
+            end
+          end
+
+          context 'when "hide_default_column_types" is "skip"' do
```

Comment:
Reviewing this is interesting. I think there's a drift between test code and what we display in options from `lib/annotate/parser.rb`